test/session-pool.js

```diff
@@ -16,2022 +16,1575 @@
+          assert.strictEqual(sessionPool.options.maxWait, 50);
+          assert.strictEqual(sessionPool.options.idlesAfter, 50);
+          assert.strictEqual(sessionPool.options.maxReads, 10);
+          assert.strictEqual(sessionPool.options.maxWrites, 5);
+          assert.strictEqual(sessionPool.options.minReads, 0);
+          assert.strictEqual(sessionPool.options.minWrites, 0);
+        });
 
-      it('should throw when writes is greater than 1', function() {
-        assert.throws(function() {
-          return new SessionPool(DATABASE, {writes: 50});
-        }, /Write percentage should be represented as a float between 0\.0 and 1\.0\./);
-      });
-    });
+        it('should not override user options', function() {
+          return sessionPool.close().then(function() {
+            sessionPool = new SessionPool(DATABASE, {acquireTimeout: 1});
+            assert.strictEqual(sessionPool.options.acquireTimeout, 1);
+            assert.strictEqual(
+              sessionPool.readPool._config.acquireTimeoutMillis,
+              1
+            );
+            assert.strictEqual(
+              sessionPool.writePool._config.acquireTimeoutMillis,
+              1
+            );
+          });
+        });
 
-    it('should set isOpen to false', function() {
-      assert.strictEqual(sessionPool.isOpen, false);
-    });
+        it('should apply defaults', function() {
+          return sessionPool.close().then(function() {
+            sessionPool = new SessionPool(DATABASE);
+            assert.strictEqual(sessionPool.options.acquireTimeout, Infinity);
+            assert.strictEqual(sessionPool.options.concurrency, 10);
+            assert.strictEqual(sessionPool.options.maxWait, 50);
+            assert.strictEqual(sessionPool.options.idlesAfter, 50);
+            assert.strictEqual(sessionPool.options.max, 100);
+            assert.strictEqual(sessionPool.options.writes, 0);
+            assert.strictEqual(sessionPool.options.min, 0);
+            assert.strictEqual(sessionPool.options.maxReads, 100);
+            assert.strictEqual(sessionPool.options.maxWrites, 0);
+            assert.strictEqual(sessionPool.options.minReads, 0);
+            assert.strictEqual(sessionPool.options.minWrites, 0);
+          });
+        });
 
-    it('should localize database request functions', function() {
-      assert.strictEqual(sessionPool.request_, DATABASE.request);
-      assert.strictEqual(sessionPool.requestStream_, DATABASE.requestStream);
-    });
+        it('should override DEFAULT for writes', function() {
+          return sessionPool.close().then(function() {
+            sessionPool = new SessionPool(DATABASE, {writes: 0.5});
+            assert.strictEqual(sessionPool.options.acquireTimeout, Infinity);
+            assert.strictEqual(sessionPool.options.concurrency, 10);
+            assert.strictEqual(sessionPool.options.maxWait, 50);
+            assert.strictEqual(sessionPool.options.idlesAfter, 50);
```

Comment:
Can we remove any assertions that do not pertain to the test? Seems like concurrency/etc. shouldn't be influenced by the `writes` option.

Reply:
We need the assertions to make sure that when we are trying to change one option then only that option is changed and the default value of other options are not affected